fix(common): add params and report progress to put

ajitsinghkaler opened the pull request:
When response type is Json the put sigbnature did not have report progress and params key. This makes it difiicult to type check added them to the signature

Fixes #23600

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: #23600


## What is the new behavior?


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

jelbourn reviewed (approved):
LGTM

Reviewed-for: public-api

ajitsinghkaler commented:
Do we need a google3 presubmit on this??

petebacondarwin reviewed (changes requested):
This seems like a perfectly reasonable fix unless someone can explain why this particular overload did not have these options? It looks like a simple oversight to me.

There are some typos in the commit message. Can you change it to:

```
fix(common): add `params` and `reportProgress` options to `HttpClient.put()` overload

When the response type is JSON, the `put()` overload signature did not have `reportProgress`
and  `params` options. This makes it difficult to type-check this overload.

This commit adds them to the overload signature.

Fixes #23600
```

ajitsinghkaler committed:
fix(common): add `params` and `reportProgress` options to `HttpClient.put()` overload

When the response type is JSON, the `put()` overload signature did not have `reportProgress`
and  `params` options. This makes it difficult to type-check this overload.

This commit adds them to the overload signature.

Fixes #23600

ajitsinghkaler commented:
@petebacondarwin changed the message

petebacondarwin reviewed (approved):
Reviewed-for: public-api

petebacondarwin reviewed (approved):
Reviewed-for: public-api-approvers

atscott reviewed (approved):
Reviewed-for: public-api

atscott reviewed (approved):
Reviewed-for: fw-core

AndrewKushnir reviewed (approved):
Reviewed-for: public-api

AndrewKushnir commented:
FYI, the remaining group that needs to give an LGTM is `fw-http` (@alxhub).

ajitsinghkaler commented:
@AndrewKushnir is there a problem with the presubmit. Should I change something in this

AndrewKushnir commented:
Hi @ajitsinghkaler, I've [started a new presubmit](http://test/OCL:331798203:BASE:331798256:1600190673218:68f6e740) and will let you know how it goes. Meanwhile, could you please rebase this PR so we re-run CI for this change with the most up-to-date version of master branch? Thank you.

ajitsinghkaler commented:
@AndrewKushnir rebased it.

AndrewKushnir commented:
Hi @ajitsinghkaler, the presubmit went well. It looks like this PR has all approvals, so I'm adding the "merge" label. Thank you.
